### PR TITLE
Bulk preview reassurance

### PR DESCRIPTION
**Pull request by amazingphilippe**
# Summary | Résumé

Implements [notification-planning/192](https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/192)

# Affected templates:

- ok.html
- error-rows.html
- error-columns.html

To access these templates:
1. On an email and a sms template
2. Add multiple recipients and add one of the following csv
    - A good csv
    - A csv with bad emails or bad phone numbers
    - A csv with the wrong headers
3. Review the pages. List should be limited to the first 10 entries. 
4. If your file is valid, you'll be able to load more entries for preview
5. If your file is invalid, you wont see a preview
6. Moving from previous to next preview should be allowed, the first email only has a "next preview" button the last preview only has a "previous preview" button

# Reviewer checklist
## Someone
- [x] Scan one of the preview page with automated tools.
- [x] Read through using voiceover

## A dev
- [x] Is the code alright?
- [x] Should we add new tests?
- [x] All tests pass

## A designer
- [x] No missing translations
- [x] Matches Figma

# Known problems with the tables

to be fixed by another PR:

- Not quite responsive.
- Sticky button container is broken.
- Index rows can't be read by screen reader, but can be tabbed through.
- Index rows don't have a descriptive accessible name.

### CODE REVIEW

**Comment by sastels** [comment]
In 
> Only shows the first 10 recipients. [Show the next 10 recipients]
clicking the "Show the next 10 recipients" shows the first 20. Should that just show 11 - 20?

File: app/main/views/send.py [hunk 1]
```diff
@@ -657,7 +657,7 @@ def _check_messages(service_id, template_id, upload_id, preview_row, letters_as_
         contents,
         template_type=template.template_type,
         placeholders=template.placeholders,
-        max_initial_rows_shown=50,
+        max_initial_rows_shown=int(request.args.get("show") or 10),
         max_errors_shown=50,
```
[the remark this answers is not in the document]
Comment (amazingphilippe) [reply app/main/views/send.py:661]: Good point, I haven't tested with that many errors. Also not sure there is a link to load more errors.

**Comment by amazingphilippe** [comment]
> Should that just show 11 - 20?

Interesting question. We could tweak the words if needed. On an interaction perspective, the goal is to load 10 more and add them to the visible list. Maybe that could be it: "Show 10 more rows"

**Comment by sastels** [comment]
whoops, did not mean to close. Rather to comment that re how many rows to show when "10 more" is clicked, we could go with what we have now and do a bit of testing to see if that's confusing or if it's fine.

**Review by adriannelee — CHANGES_REQUESTED** [review]
Email, good list:

- Are the table row numbers meant to be interacted with? They don't seem to go anywhere. There are also aria-hidden="true" elements in the recipients list that contain focusable elements (should have tabindex='-1' or be removed from the DOM)
- With VoiceOver on Chrome, it seems like the user should be able to skip interacting with the table if they want to skip down to the next section. Right now, they have to read the whole table to get to the next section. (Maybe related to the aria-hidden comment above?)

**Review by adriannelee — CHANGES_REQUESTED** [review]
Email, bad file: 
- There's a "Back to top" button that doesn't seem to go anywhere? 
- There is an invisible element called "Add recipients" that can be read by a screen reader, just before "Re-upload your file"'
- Not sure if this is related to the comment above: Does this element have multiple label elements?
` <input class="file-upload-field" id="file" name="file" required="" type="file">`
- "Re-upload your file" is read twice, once as a button and another time as a text element
- The "Re-upload your file" button also reads that it requires an "invalid" file. I'm not sure what this means.
Screen Shot 2022-03-29 at 12 24 26 PM (2)

- Phone numbers in the sheet are read as "one billion eight hundred one million .. " .. should it read more simply?
- The user has to read the whole table in order to skip to the next section.

**Comment by adriannelee** [comment]
Email, no headers:
- Not sure if this is what the page should look like (with the email preview)? I got the same result as uploading the 'bad phone' file
screencapture-notification-feat-reass-khsroj-herokuapp-services-464521de-2ee6-4aae-87ff-ba08c68adc72-55f3896c-cd32-4428-a1e3-78a81ef2d249-check-d64c9b94-a53e-436f-812d-33a0ec6d7c92-2022-03-29-16_03_00
(So the same a11y issues were found on this page)

**Comment by amazingphilippe** [comment]
> There's a "Back to top" button that doesn't seem to go anywhere?

Yes... We can address this in another PR

> There is an invisible element called "Add recipients" that can be read by a screen reader, just before "Re-upload your file"'
> Not sure if this is related to the comment above: Does this element have multiple label elements?

Yes! There are too many labels for the file input. We have an issue open (or in airtable) to fix file inputs. The fix would improve this page too. 

> "Re-upload your file" is read twice, once as a button and another time as a text element

Was this on chrome or safari? And it also seems that the visible button and actual button don't match. One says "Add recipents", while the button shows "Re-upload your file"

> The "Re-upload your file" button also reads that it requires an "invalid" file. I'm not sure what this means.
This is also a chrome and voiceover issue. Safari doesn't have this issue. 

> Phone numbers in the sheet are read as "one billion eight hundred one million .. " .. should it read more simply?

Probably worth trying!

> The user has to read the whole table in order to skip to the next section.

This is okay when navigating with arrows. Users would  use different techniques like navigating by regions or headings to skip sections of a page. 

> VoiceOver on Chrome will read each cell item 3 times in the table.

Issue not present with voiceover in Safari.

**Comment by adriannelee** [comment]
>> "Re-upload your file" is read twice, once as a button and another time as a text element

> Was this on chrome or safari? And it also seems that the visible button and actual button don't match. One says "Add recipents", while the button shows "Re-upload your file"

I only tested with VoiceOver with Chrome. When I try with Safari with VoiceOver, it doesn't read "Re-upload your file" twice. I heard the same thing as you - "Add recipients" instead of "Re-upload your file." It seems like for both Chrome and Safari, "Re-upload your file" shouldn't be read as a text element and it should be read once as a button.

> This is okay when navigating with arrows. Users would use different techniques like navigating by regions or headings to skip sections of a page.

True! I didn't think about this.

**Review by sastels — APPROVED** [review]
LGTM if @adriannelee is happy

**Review by adriannelee — APPROVED** [review]

**Pull request merged**